From b71cc3640855f8638e40bc178abd0a4fae3a4d4c Mon Sep 17 00:00:00 2001
From: Gabriele M <moto.falcon.git@gmail.com>
Date: Wed, 17 Feb 2016 10:03:37 +0100
Subject: [PATCH 064/194] Snap: Don't crash if user saved preference is not
 valid

If the user sets some preference value that is later removed, Snap will
crash because of on an ArrayIndexOutOfBoundsException. Make sure this
doesn't happen.

Change-Id: I629acebaf762f49f05dc1bfa0542f11979ae269d
---
 src/com/android/camera/ListPreference.java | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/com/android/camera/ListPreference.java b/src/com/android/camera/ListPreference.java
index ae37356e1..05ab10f07 100644
--- a/src/com/android/camera/ListPreference.java
+++ b/src/com/android/camera/ListPreference.java
@@ -184,7 +184,11 @@ public class ListPreference extends CameraPreference {
     }
 
     public String getLabel() {
-        return mLabels[findIndexOfValue(getValue())].toString();
+        int index = findIndexOfValue(getValue());
+        if (index < 0) {
+            return findSupportedDefaultValue();
+        }
+        return mLabels[index].toString();
     }
 
     protected void persistStringValue(String value) {
-- 
2.17.1

